aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterDeleteTest.java

```diff
@@ -259,19 +262,19 @@ public void deleteModelTypeWithPredicateCascades() throws DataStoreException {
         }
 
         // Observe deletions
-        Set<String> deleted = new HashSet<>();
-        adapter.observe()
+        TestObserver<String> deleteObserver = adapter.observe()
                 .filter(change -> StorageItemChange.Type.DELETE.equals(change.type()))
                 .map(StorageItemChange::item)
-                .subscribe(model -> deleted.add(model.getId()));
+                .map(Model::getId)
+                .test();
 
         // Triggers a delete of all blogs.
         // All posts will be deleted by cascade.
         adapter.delete(Blog.class, QueryPredicates.all());
 
-        // Assert that cascaded deletions are observed.
-        assertEquals(12, deleted.size());
-        assertEquals(expected, deleted);
+        // Assert 3 blogs and 9 posts are deleted.
+        deleteObserver.assertValueCount(12);
+        assertEquals(expected, new HashSet<>(deleteObserver.values()));
```

Comment:
There is actually an `assertValueSetOnly` which may be appropriate here.

It does **not** work like a `HashSet`, de-duplicating values. It **does** ignore the order in which events arrived on the stream, which I believe is the real reason that a Set was being used in the test to begin with.

Might be a little cleaner.

Actually come to think of it, this would remove the need for the count check, as well.
```suggestion
        deleteObserver.assertValueSetOnly(expected);
```

Reply:
`assertValueSetOnly` was deprecated and is no longer supported by rxjava3 :(